File: SchoolWebApp-API/Project.Core/Entities/Academics/ExamResult.cs
﻿using SchoolWebApp.Core.Entities.Shared;
using SchoolWebApp.Core.Entities.Students;
using System.ComponentModel.DataAnnotations;

namespace SchoolWebApp.Core.Entities.Academics
{
    public class ExamResult : Base
    {
        [Required(ErrorMessage = "Enter the examination score")]
        [Display(Name = "Examination score")]
        public float Score { get; set; }

        public int StudentId { get; set; }
        public Student? Student { get; set; }
        public int ExamId { get; set; }
        public Exam? Exam { get; set; }
    }
}

File: SchoolWebApp-Client/src/app/cbe/exams/models/exam-result.ts
import {ResourceModel} from '@/core/models/ResourceModel';

export class ExamResult extends ResourceModel<ExamResult> {
    public score?: number;
    public description?: string;
    public studentId?: number;
    public student?: any;
    public examId?: number;
    public exam?: any;

    constructor(model?: Partial<ExamResult>) {
        super(model);
    }
}
